File: app/src/main/java/io/mgba/Data/Database/Converters/GameConverter.java
package io.mgba.Data.Database.Converters;

import android.arch.persistence.room.TypeConverter;

import java.io.File;

public class GameConverter {
    @TypeConverter
    public static File fromStringToFile(String value) {
        return value == null ? null : new File(value);
    }

    @TypeConverter
    public static String fromFileToString(File value) {
        return value == null ? null : value.getAbsolutePath();
    }

}

File: app/src/main/java/io/mgba/data/database/converters/GameConverter.kt
package io.mgba.data.database.converters

import java.io.File

import androidx.room.TypeConverter

object GameConverter {
    @TypeConverter
    fun fromStringToFile(value: String?): File? {
        return if (value == null) null else File(value)
    }

    @TypeConverter
    fun fromFileToString(value: File?): String? {
        return value?.absolutePath
    }

}
